tremor-script/src/interpreter/imut_expr.rs

```diff
@@ -472,26 +472,53 @@ impl<'script> ImutExpr<'script> {
     where
         'script: 'event,
     {
-        let lval = stry!(expr.lhs.run(opts, env, event, state, meta, local)).try_as_bool()?;
+        let lhs = stry!(expr.lhs.run(opts, env, event, state, meta, local));
+        let lval = lhs.try_as_bool().map_err(|e| {
+            ErrorKind::InvalidBinaryBoolean(expr.extent(), expr.extent(), expr.kind, e.got, None)
```

Comment:
Inner should be expr.lhs.extend() here, and expr.rhs.extent() in the lower errors. We want the error to show  the entire expression but point to the part that's wrong.